### PR TITLE
add Salesforce SOAP connector

### PR DESCRIPTION
**Pull request by Arshardh**

### CODE REVIEW

File: salesforcesoap/src/org/wso2/ballerina/connectors/salesforcesoap/salesforcesoapConnector.bal [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+            map n = {"ns":"urn:partner.soap.sforce.com"};
+            sessionId = xmls:getString(loginResponse, "/ns:loginResponse/ns:result/ns:sessionId/text()", n);
+            session = `<urn:SessionHeader xmlns:urn="urn:partner.soap.sforce.com"><urn:sessionId>${sessionId}</urn:sessionId></urn:SessionHeader>`;
+            serverUrl = xmls:getString(loginResponse, "/ns:loginResponse/ns:result/ns:serverUrl/text()", n);
+        }
+        headers[headerCount] = session;
+        xml payload = `<urn:create xmlns:urn="urn:partner.soap.sforce.com" xmlns:urn1="urn:sobject.partner.soap.sforce.com">
+                           <urn:sObjects>
+                               <urn1:type>${type}</urn1:type>
+                               <urn1:Name>${name}</urn1:Name>
```
Comment (hariss63) [inline salesforcesoap/src/org/wso2/ballerina/connectors/salesforcesoap/salesforcesoapConnector.bal]: The connector should handle all the fields of sObject. The fields differ from sObject to sObject. We need to think to handle those in create, update, upsert actions.

Comment (Arshardh) [inline salesforcesoap/src/org/wso2/ballerina/connectors/salesforcesoap/salesforcesoapConnector.bal]: As we discussed in the yesterday's meeting, we will be accepting mandatory parameters for an action as action parameters and the remainder as a struct. Once this approach is finalized, we can request a struct from the user with all the remaining parameters. For the today's release we will include only the mandatory parameters.

Comment (hariss63) [inline salesforcesoap/src/org/wso2/ballerina/connectors/salesforcesoap/salesforcesoapConnector.bal]: Yes, we will do it later with the struct concept. But the Name is the mandatory one for Account sObject but when you choose another type of sObject then the mandatory params are differ.

Comment (Arshardh) [inline salesforcesoap/src/org/wso2/ballerina/connectors/salesforcesoap/salesforcesoapConnector.bal]: I added a map to obtain the required fields from the user

**Review by hariss63 — APPROVED** [review]

**Pull request merged**